presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestSimplifyExpressions.java

```diff
@@ -118,11 +123,14 @@ private static void assertSimplifies(String expression, String expected)
         Expression actualExpression = rewriteIdentifiersToSymbolReferences(SQL_PARSER.createExpression(expression));
         Expression expectedExpression = rewriteIdentifiersToSymbolReferences(SQL_PARSER.createExpression(expected));
         assertEquals(
-                normalize(simplifyExpressions(actualExpression)),
+                normalize(simplifyFilterExpressions(actualExpression)),
+                normalize(expectedExpression));
+        assertEquals(
+                normalize(simplifyJoinExpressions(actualExpression)),
```

Comment:
Do we have an expression already in the test cases that verifies the scenario you presented in the query?